### ISSUE TITLE
azuread_user support for givenName and surname attributes

### ISSUE THREAD

**Issue by lelama** [terraform-providers/terraform-provider-azuread#301]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

First Name (or Given Name) and Last Name (or Surname) attributes are required by some applications (e.g. [AWS SSO](https://docs.aws.amazon.com/singlesignon/latest/userguide/azure-ad-idp.html)) using Azure AD as IdP in order for SCIM provisioning to succeed. Please add support for these 2 attributes.

### New or Affected Resource(s)

* azuread_user

### Potential Terraform Configuration

```hcl
resource azuread_user newuser {
     user_principal_name  = "new.user@example.com"
     display_name = "New User"
     first_name = "New"
     last_name = "User"
}
```

### References

* similar to #281 which is adding attributes from "Job Info" and "Contact Info" sections of the Azure AD User object.

**Comment by manicminer** [comment]
Hi @lelama, thanks for requesting these. I'll look at including givenName and surname attributes in the next release of the provider.

**Comment by manicminer** [comment]
This was implemented in #351
